### PR TITLE
Added service type to all istio components.

### PR DESCRIPTION
**Pull request by gyliu513**
Fixed https://github.com/istio/istio/issues/5944

/cc @sdake @sbezverk @costinm

### CODE REVIEW

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -403,7 +419,7 @@ tracing:
     tag: 1.5
   service:
     name: http
-    type: ClusterIP
+    type: LoadBalancer #change to NodePort, ClusterIP or LoadBalancer if need be
```
Comment (objectiser) [inline install/kubernetes/helm/istio/values.yaml]: This should be `ClusterIP` for the zipkin service.

Need to add a `uiType: LoadBalancer` for the UI (tracing) service.

File: install/kubernetes/helm/istio/charts/tracing/templates/service.yaml [hunk 2]
```diff
@@ -31,12 +31,12 @@ items:
       release: {{ .Release.Name }}
       heritage: {{ .Release.Service }}
   spec:
+    type: {{ .Values.service.type }}
```
Comment (objectiser) [inline install/kubernetes/helm/istio/charts/tracing/templates/service.yaml]: As per change below, this should become `.Values.service.uiType` - as distinct from the service type of the zipkin service.

Comment (gyliu513) [reply install/kubernetes/helm/istio/charts/tracing/templates/service.yaml]: Ah, I see, then I'd like to keep here as unchanged, just hardcode the service type as loadbalancer for now.

**Comment by costinm** [comment]
Closing this - please reopen after discussion on the mailing list, involving security as well.

**Comment by gyliu513** [comment]
Posted at https://groups.google.com/forum/#!topic/istio-environments/V4Uml3R4AZU in mail list @costinm

**Pull request closed without merging**